PivotGridDataSource: update selector (#7434)
DevExpress/devextreme-documentation: api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/selector.md

---
---
##### shortDescription
Specifies a function that combines field values into groups.

---

#include btn-open-demo with {
    href: "https://js.devexpress.com/Demos/WidgetsGallery/Demo/PivotGrid/SimpleArray/"
}

[note]

- The **selector** function cannot be used with an an [XmlaStore](/api-reference/30%20Data%20Layer/XmlaStore '/Documentation/ApiReference/Data_Layer/XmlaStore/') or [remote operations](/api-reference/30%20Data%20Layer/PivotGridDataSource/1%20Configuration/remoteOperations.md '/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/#remoteOperations'). If you use remote operations, create a calculated data field on the server and [bind a pivot grid field to it](/api-reference/30%20Data%20Layer/PivotGridDataSource/1%20Configuration/fields/dataField.md '/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#dataField').
- The `this` keyword refers to the field's configuration.

[/note]

> The **selector** function takes a data object and returns a field value. You can use **selector** as an alternative to the [groupInterval](/api-reference/30%20Data%20Layer/PivotGridDataSource/1%20Configuration/fields/groupInterval.md '/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#groupInterval') property.

Reviewer: ```suggestion
The **selector** function takes a data object and returns a field value. You can use **selector** instead of the [groupInterval](/api-reference/30%20Data%20Layer/PivotGridDataSource/1%20Configuration/fields/groupInterval.md '/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#groupInterval') property.
```
